### PR TITLE
[monitoring] Remove trickster

### PR DESCRIPTION
**Pull request by Regrau**
## Description
This PR removes trickster from some modules.

## Why do we need it, and what problem does it solve?
Trickster and the aggregation proxy are solving the same problem. Hence removing trickster
and using the aggregation proxy helps remove redundancy.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: prometheus
type: chore
summary: Remove Trickster.
impact: Prometheus will restart.
---
section: prometheus-metrics-adapter
type: chore
summary: Use aggergating proxy instead of Trickster.
impact: prometheus-metrics-adapter will restart.
```

### CODE REVIEW

**Review by z9r5 — APPROVED** [review]

File: modules/110-istio/templates/kiali/rbac-for-us.yaml [hunk 1]
```diff
@@ -95,10 +95,6 @@ rules:
   - tokenreviews
   verbs:
   - create
-- apiGroups: ["apps"]
```
Comment (vladimirGuryanov) [inline modules/110-istio/templates/kiali/rbac-for-us.yaml:98]: If you remove only the RBAC for Trickster, Kiali won't work with the aggregator proxy. You need to update the RBAC settings for Kiali to work with the aggregator proxy.

File: modules/300-prometheus/templates/grafana/secret-datasources-list.yaml [hunk 2]
```diff
@@ -19,11 +19,6 @@ secureJsonData:
 apiVersion: 1
 
 deleteDatasources:
-- name: trickster
```
Comment (vladimirGuryanov) [inline modules/300-prometheus/templates/grafana/secret-datasources-list.yaml:22]: This is a list of data sources pending deletion. Generally, this list includes data sources that have already been created and are scheduled for removal. You don't need to remove the Trickster data source from here.

File: modules/300-prometheus/templates/grafana/secret-datasources-list.yaml [hunk 3]
```diff
@@ -37,14 +32,6 @@ deleteDatasources:
   {{- end }}
 
 datasources:
-- name: main-old
```
Comment (vladimirGuryanov) [inline modules/300-prometheus/templates/grafana/secret-datasources-list.yaml:40]: Needs to be moved to the 'Deleted' section, not deleted entirely.

File: modules/300-prometheus/templates/grafana/secret-datasources-list.yaml [hunk 4]
```diff
@@ -54,16 +41,6 @@ datasources:
   isDefault: true
   {{- include "authorization_config" (list . (include "main_scrape_interval" . )) | nindent 2 }}
 
-  {{- if ne (int .Values.prometheus.longtermRetentionDays) 0 }}
```
Comment (vladimirGuryanov) [inline modules/300-prometheus/templates/grafana/secret-datasources-list.yaml:57]: We cannot delete this data source. If we do, users will lose the ability to retrieve data from Prometheus LTS. I suggest updating the aggregation proxy and adding a new server group for access to Prometheus LTS, similar to how it was configured in Trickster. Finally, we should update this data source accordingly.

**Review by vladimirGuryanov — CHANGES_REQUESTED** [review]

**Review by apolovov — APPROVED** [review]

**Pull request closed without merging**